test/orbit_determination.jl

```diff
@@ -0,0 +1,303 @@
+        @test all( norm.(sol.fwd.x, Inf) .< 2 )
+        @test isempty(sol.t_fwd) && isempty(sol.x_fwd) && isempty(sol.g_fwd)
+        # Vector of residuals
+        @test length(sol.res) == 21
+        @test count(outlier.(sol.res)) == 2
+        # Least squares fit
+        @test sol.fit.success
+        @test all( sigmas(sol) .< 5e-4 )
+        @test nrms(sol) < 0.25
+        # Scalig factors
+        @test all(sol.scalings .< 8e-7)
+        # Compatibility with JPL
+        JPL = [0.7673358221902306, 0.6484904294813807, 0.2932331617634889,
+               -0.011023358761553661, 0.015392684491034429, 0.006528836324700449]
+        @test all(abs.(sol() - JPL) ./ sigmas(sol) .< 0.1)
+    end
+
+    @testset "Interesting NEOs" begin
+
+        # 2014 AA hit the Earth around January 2, 2014, 02:49 UTC
+
+        # Fetch optical astrometry
+        radec = fetch_radec_mpc("designation" => "2014 AA")
+
+        # Parameters
+        params = NEOParameters(coeffstol = Inf, bwdoffset = 0.007, fwdoffset = 0.007)
+
+        # Orbit Determination
+        sol = orbitdetermination(radec, params)
+
+        # Values by February 24, 2024
+
+        # Vector of observations
+        @test length(radec) == 7
+        @test numberofdays(radec) < 0.05
+        # Orbit solution
+        @test isa(sol, NEOSolution{Float64, Float64})
+        # Tracklets
+        @test length(sol.tracklets) == 1
+        @test sol.tracklets[1].radec[1] == radec[1]
+        @test sol.tracklets[end].radec[end] == radec[end]
+        @test issorted(sol.tracklets)
+        # Backward integration
+        @test datetime2days(date(radec[1])) > sol.bwd.t0 + sol.bwd.t[end]
+        @test all( norm.(sol.bwd.x, Inf) .< 2 )
+        @test isempty(sol.t_bwd) && isempty(sol.x_bwd) && isempty(sol.g_bwd)
+        # Forward integration
+        @test datetime2days(date(radec[end])) < sol.fwd.t0 + sol.fwd.t[end]
+        @test all( norm.(sol.fwd.x, Inf) .< 1e9 )
+        @test isempty(sol.t_fwd) && isempty(sol.x_fwd) && isempty(sol.g_fwd)
+        # Vector of residuals
+        @test length(sol.res) == 7
+        @test iszero(count(outlier.(sol.res)))
+        # Least squares fit
+        @test sol.fit.success
+        @test all( sigmas(sol) .< 1e-3 )
+        @test nrms(sol) < 0.13
+        # Scalig factors
+        @test all(sol.scalings .< 1e-5)
+        # Compatibility with JPL
+        JPL = [-0.17932853771087842, 0.8874166708545763, 0.38414497114153867,
+               -0.01755788350351527, -0.005781328974619869, -0.0020073946363600814]
+        @test all(abs.(sol() - JPL) ./ sigmas(sol) .< 0.3)
+
+        # 2008 TC3 entered the Earth's atmosphere around October 7, 2008, 02:46 UTC
+
+        # Fetch optical astrometry
+        radec = fetch_radec_mpc("designation" => "2008 TC3")
+
+        # Parameters
+        params = NEOParameters(coeffstol = Inf, bwdoffset = 0.007, fwdoffset = 0.007)
+
+        # Observations with <1" weight
+        idxs = findall(x -> x < 1, w8sveres17.(radec))
```

Comment:
```suggestion
        idxs = findall(x -> w8sveres17(x) < 1, radec)
```
(avoids creating an intermediate array)

Reply:
Done.